src/components/webview/repository.ts

```diff
@@ -0,0 +1,84 @@
+import { APIState } from '../../../git';
+import { API, Repository } from '../../../git';
+import { assertsNeitherNullOrUndefined } from '../../utilities';
+
+const branchNameFromStr = (str: string): string => {
+	let branchName = str.toLowerCase();
+
+	branchName = branchName.replace(/\s+/g, '-');
+
+	branchName = branchName.replace(/[^a-z0-9-]/g, '-');
+
+	branchName = branchName.replace(/--+/g, '-');
+
+	branchName = branchName.replace(/^-+|-+$/g, '');
+
+	if (branchName.length > 63) {
+		branchName = branchName.substr(0, 63);
+	}
+
+	if (!/^[a-z0-9]/.test(branchName)) {
+		branchName = 'x-' + branchName;
+	}
+
+	return branchName;
+};
+
+export class UninitializedError extends Error {}
+
+export class RepositoryService {
+	__repo: Repository | null = null;
+
+	constructor(private readonly __gitAPI: API) {
+		this.__gitAPI.onDidChangeState(this.onDidChangeState);
+	}
+
+	private onDidChangeState = (state: APIState) => {
+		if (state === 'initialized') {
+			this.__repo = this.__gitAPI.repositories[0] ?? null;
+		}
+	};
+
+	public getAllBranches = async () => {
```

Comment:
I would not use arrow functions as methods, this is syntax sugar that is confusing (at least to me).